### ISSUE TITLE
Peanut Plug power reporting issues

### ISSUE THREAD

**Issue by hjbreese** [pakmanwg/smartthings-peanut-plug#1]
I have 2 old and 4 new Peanut Plugs (PP).  I purchased the 2 old plugs last year and used my Almond router to update the firmware to the newest (at that time) version.  I am using your DH for these devices.  They work just fine and report power usage.  The only issue with them is that there are a LOT of digits on the right side of the decimal point instead of just a number, decimal point, and a single digit.

The four new PP reported that they already had the latest firmware installed when I paired them with my Almond router and went through the firmware update procedure.  I then paired them with my ST v2 hub.  I use the Classic ST phone app on my Android phone.  I am using your device handler for these PP.  It's the same one as I use for the original two PP.  Power reporting does not work with the four new plugs.

Do you know if the firmware for these PP has been changed by Securifi so that they will now not report power usage with your device handler?  This is the only thing I can think of that would be causing the issue I am seeing.  If so, will you please upgrade your DH for these devices and also change the format of the displayed number to be nn.n (where "n" is a number reporting power usage.)  The rounding issue is in both the current usage and accumulated usage.

Thanks!
Hugh

**Comment by ChitownM2** [comment]
Ordered some new PP this week and am having the same issue. I don't have an almond router, but I ordered them off amazon so I'm assuming they are pretty new stock and have the latest firmware.

**Comment by hjbreese** [comment]
I am still waiting to hear back from Securifi to find out if they updated the PP firmware power reporting.  Perhaps if more people contact them to find out about this, it might help get the ball rolling?

**Comment by ChitownM2** [comment]
I'm guessing without an almond hub I can't even view the firmware that is currently on the peanut plugs. Are you able to see the firmware revision number when you attempt to update? 

In order to troubleshoot this issue I'm pretty sure he will need to know the revision. For example, if you're 2 old ones were on version 3.00 & you updated them to 4.00 and everything worked, but the 4 newest ones are on 5.0 then it means securifi probably changed something that broke compatibility with the DH. If all 6 have the same firmware version then it's likely a different problem.

**Comment by hjbreese** [comment]
ChitownM2,  My experience with this overall exercise is that the firmware on my new PP's is not the latest firmware even though they are brand new.

One of the things I don't like in general about Securifi routers is they are overly simplistic.  The only way to find some of the information is by using a telnet session into the router.  Good luck finding documentation on that.  Finding out the currently installed firmware version is not available in the GUI interface.  When upgrading the firmware, the GUI interface only shows that an update is available.  Again, no version number.

I had a support tech call me about this issue today.  Here is the result of what happened:

The support rep tried to no avail to manually update the firmware in the PP's.  He was unable to do this and scheduled a upper level support appointment for tomorrow.  When he disconnected from his remote access session, my Almond router was totally unresponsive, so he had accidentally scrambled something in the Almond during his telnet session.  I ended up resetting my Almond 2015 router back to factory default settings and reconfiguring it again.  It's a good thing that I only use it for updating the firmware in PP's.  After resetting the router, I was able to update all of my new PP's to the latest firmware.

Now all my new PP's are reporting power usage (Yay!)  The only issue now is that the decimal amounts displayed in the tiles are much too accurate.  What I would like to see are values rounded to the nearest tenth in the two tiles.  The Current Usage tile is currently displaying 10 digits to the right of the decimal point (this occasionally displays a number rounded to a tenth for a fraction of a second before it switches back to displaying 10 digits to the right of the decimal point.)  The Accumulated Usage tile displays 23 digits to the right of the decimal point (I have never seen this number display a "rounded to the tenth" of a kWh value.)
